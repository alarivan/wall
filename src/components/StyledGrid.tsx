import styled from 'styled-components';
import { TColor } from '../types';

interface StyledGridProps {
  readonly rows: number;
  readonly columns: number;
  readonly size: number;
  readonly preview: boolean;
  readonly background: TColor;
}

function getBorderColor({ preview }: StyledGridProps): string {
  return preview ? 'black' : 'black white white black';
}

const StyledGrid = styled.div<StyledGridProps>`
  display: grid;
  justify-content: start;
  grid-gap: 0;
  grid-template-columns: repeat(
    ${props => props.columns + ', ' + props.size}px
  );
  grid-template-rows: repeat(${props => props.rows + ', ' + props.size}px);

  border-style: solid;
  border-width: 1px;
  border-color: ${getBorderColor};
  background: ${({ background }) => background};
`;

export default StyledGrid;
