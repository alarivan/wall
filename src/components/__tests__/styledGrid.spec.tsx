import React from 'react';
import { render, cleanup } from '@testing-library/react';
import StyledGrid from '../StyledGrid';

afterEach(cleanup);

const props = {
  columns: 2,
  rows: 2,
  size: 10,
  preview: false,
};

describe('StyledGrid', () => {
  it('renders correctly', () => {
    const { asFragment, container } = render(<StyledGrid {...props} />);

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-columns',
      'repeat( 2,10px )',
    );

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-rows',
      'repeat(2,10px)',
    );

    expect(container.firstChild).toHaveStyleRule(
      'border-width',
      '1px 0px 0px 1px',
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with preview', () => {
    const { asFragment, container } = render(
      <StyledGrid {...props} preview={true} />,
    );

    expect(container.firstChild).toHaveStyleRule('border-width', '1px');

    expect(asFragment()).toMatchSnapshot();
  });
});
