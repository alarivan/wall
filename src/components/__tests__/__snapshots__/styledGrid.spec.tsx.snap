// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledGrid renders correctly 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  grid-gap: 0;
  grid-template-columns: repeat( 2,10px );
  grid-template-rows: repeat(2,10px);
  border-style: solid;
  border-width: 1px;
  border-color: black white white black;
  background: red;
}

<div
    class="c0"
    rows="2"
    size="10"
  />
</DocumentFragment>
`;

exports[`StyledGrid renders correctly with preview 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  grid-gap: 0;
  grid-template-columns: repeat( 2,10px );
  grid-template-rows: repeat(2,10px);
  border-style: solid;
  border-width: 1px;
  border-color: black;
  background: red;
}

<div
    class="c0"
    rows="2"
    size="10"
  />
</DocumentFragment>
`;
